### ISSUE TITLE
Add line break before start of argument list 

### ISSUE THREAD

**Issue by alanlomeli** [fsprojects/fantomas#2335]
Issue created from [fantomas-online](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAXAngB0WAYgJaIA28AcrOvrAK4B28AogB7SLbqGz0A8A5ABUAfAApYAIwBWKIQBowAM2JkUAZ3QAnQvQDmCtbmgAZHQGt1WnboCUYALwAdehAg6AFom3owAZQDuAIa6ul6%2B0J4AtoEACoGaaois7Jzc9KLOrllghtr06IpgjpAA8tKI0D4AsgCqvoJgcPmBOkoq8GAABgCkap1gor6IeL1KsJpgkeN4SOgtJGo2AHSOzmVSFego3QCCxZnZh8qkHYYVpvRmYJJSNpBy4FCJ6Fx6aqhgANqPEKCHkAASLC4D6QHToe4-VyAgBugRItGQaG%2BhwgAAYHqioABJRiIfK%2BQgAL2QmNRABYoRAALpQgC%2BZN%2BVKgQJwSKg4Mh-wBcIR7JRqIAjIzspAqoEWBdEMZ8bp0O4uUKAExo5m0rIMqF-UWskFoSD4%2BAlRRS3xYEik5mw%2BGIj4Cw5KkVZSBMRjGqWK-4kRSQNX0kXa5269mQCSwWAkT3Qnk2-nMiAAZid0NxiU01B08PIiH8JB0lqxWkRfo1AatwdBYYjUYg1r5dvjYHJydrvmwgXYACFEIppnFNIFIoh0F4a65FPDEiXXJqsoHo8CQ1XI8m67bkY2AKwtqBtjuIbu9zTS2D%2BLzQQKJXEw2AXrg8McQCcLRDTiCz1zz2sV-XLsdruMsQANh3SA9y7HtphqbBcE0C8r3oG87zSR8lEnV9DnVGcy25RdK3DFdy15dcvkbAB2UDwIPSDjwAYRIS81FongNE0WhKnGVDnynTD-S1cs8N-Aj-xjesNyxAAOSj2wgo9ECqRBIgkUcd24jDsiw98cJ1QSoD-VdRJI%2B1sgATmk-dD2mZiSAfVT0LfMAPyZXC2Xw6sDOIwChQxRswJkxAdkUEdNGYyJolQot1KyTTHO0oNdNDYSPNjBssUFYVfKoyzjyGSJCDgGz6C4%2BzeNLfiXL1PSkqIlLxOVcz2EC4LcvyiNbMbSKHKcsAvxZBL9JqsTSLSpNMv8nZNDoRgABFSEIPLgoitiotcGLut6wEEs5ZKhuMrJBWbXzxRYbFFEETx6CYF9fHccZ0AAdUIeB5VQ8lVVK7Dyp01z9W2wajMbQVtyOiVcWUFgSlgwJ0HGVhsGPNQ1BQndN3ejS%2BLnASfo5fIRM81KhRAkGWAAJQqcZ4Ee56FR3N6urihdsbBXGdoBtKKOJsm4E0ChaCUrxjWxEdIneHdBXpr74qZyJaBILg83oRAaE0aIXi8QRXNZrzDkFKTfK5imqll%2BX82V1XFtAiJ4g7YKAH1-Cel6JcxiqQz%2B7l8bqnWzOJiaB0wEpNFMDQqZe2m0eijHPyxyrmYhLWCYdHysTFCU-cCAOg8IDRyD55TNEF4XRcB53o9d0EZblwgFaV8ZzfVzX-u17IlQylP08z4P0CNqua7N6GLd8q2B0qLx7cdhVS%2Bc77Y-dnVPeG1ElUdYmADVY07HR4GsUOacbCSI9WqPp6l2eWabxOW9GlPjvwBhKjSTfGB3ifXsPmlj56mO3fPj3asXh0h0b4SmmlQAA4sOOGCMkY8F3qhVGU8v7lyEu5C%2BXsW7Axvsbau%2BZOw2WgGYTsI8zDDjUCUegvhByIGsnzIqO5OofS0pLRmscBp-12o2JURMU7ZlzLg4cZ58Qa1wLNZQ9BCD3noDsRgCl%2BYJGKi%2BRBG0fxVVQewtmS8ObtzzLoegd96APx4ISXR0NaDHkELAXESBmiSIUTxdGZUXYzyXNVdRzcshKj1to4K9BoaEBhCePQsj86iJ0BItIxcsRqSUd-NyhE3GXw8T7LBVcpTGEHBIeAgR6KwCRnoXhNc7ErQ-o4suzi4l43-ntVwCZk6ohdCwWC818RzBIAAaWGNgKxLTcREMCAYveUSSoOM%2Bk40%2BLi1HzyqY2BMbd6l4P6WYKUagJpTQoIgDQiB4A9xNorKBGyYH0EiaiBhIymFjJYRM%2BJUyOFYgTCvFOnZ4jZUQNNbO0BtB5V8SOeANRxE8FmtABiA5bF2UUYw2KzDvz9VcTcjRhwEzX3qXDLwzTmgkDNJNWgrFaDYDNJgC0S1iwQvWrE36v84XuJqUA%2BpHSODHVznI40CzLjLNQoKNE78wBrQZtCpmbDKWJJqZg%2BpmL8rdwQAWVE0SSWPGpGSMEah8BIw%2BNEukQA)

#### Code

```fsharp
type FieldNotFoundException<'T>(obj:'T, field:string, specLink:string) =
    inherit SwaggerSchemaParseException(
        sprintf "Object MUST contain field `%s` (See %s for more details).\nObject:%A"
            field specLink obj)
```
            

#### Result

```fsharp
type FieldNotFoundException<'T>(obj: 'T, field: string, specLink: string) =
    inherit SwaggerSchemaParseException(sprintf
                                            "Object MUST contain field `%s` (See %s for more details).\nObject:%A"
                                            field
                                            specLink
                                            obj)

```
            
#### Problem description

After F# 6 indentation relaxations, we can now improve indent consistency by adding a line break before the argument.
                    Check out our [Contribution Guidelines](https://github.com/fsprojects/fantomas/blob/master/CONTRIBUTING.md#bug-reports).

#### Extra information

- [ ] The formatted result breaks my code.
- [ ] The formatted result gives compiler warnings.
- [ ] I or my company would be willing to help fix this.

#### Options

Fantomas master branch at 2022-07-04T06:38:00Z - 26d45cf5a75ae44efd2bc7b3d80c660c77024a51

```fsharp
    { config with
                SpaceBeforeParameter = false
                SpaceBeforeLowercaseInvocation = false
                MultilineBlockBracketsOnSameColumn = true
                ExperimentalStroustrupStyle = true }
```

<sub>Did you know that you can ignore files when formatting from fantomas-tool or the FAKE targets by using a [.fantomasignore file](https://github.com/fsprojects/fantomas/blob/master/docs/Documentation.md#ignore-files-fantomasignore)?</sub>

**Comment by nojaf** [comment]
@yisusalanpng the problem also occurs using the default settings, so the settings you have listed don't have an influence on the problem. It is best to leave them out when reporting them.
Thank you for the report.

**Comment by nojaf** [comment]
@yisusalanpng @sergey-tihon I've raise this with the style guide, https://github.com/fsharp/fslang-design/issues/693.
Just to be on the safe side.

**Comment by nojaf** [comment]
@yisusalanpng I'm not really sure what I was going to write down as additional feedback.
So I'll add some generic pointers:

Almost every function in CodePrinter takes the `Context` record as the last argument.
In F# you can omit this parameter when defining a function as the [partial application](https://fsharpforfunandprofit.com/posts/partial-application/) will kick in.

Take the `genIdent (ident: Ident) = !-ident.idText` function for example.
In full this would be `genIdent (ident: Ident) ctx = !-ident.idText ctx`.
The definition of `(!-)`  is 
![image](https://user-images.githubusercontent.com/2621499/178312701-0ea08710-df87-48a2-8b6f-c76bfe72b276.png)

Notice the `string -> Context -> Context` type signature. This function returns a `Context`.
And take a `string` and `Context` argument. If we only pass in a string (like `!- "hey"`), this will return a function that takes `Context` and returns a `Context`.
Because the `(+>)` combines two functions, it is very similar to [>>](https://fsharp.github.io/fsharp-core-docs/reference/fsharp-core-operators.html#(%3E%3E)).
We can usually drop the last `Context` and return `Context -> Context`.

For example

```fsharp
let f (ctx:Context) : Context = !- "f" ctx // Adds WriterEvent.Write "f" to the Context
let g (ctx:Context) : Context = != "g" ctx // Adds WriterEvent.Write "f" to the Context

let combined (ctx:Context) = g (f ctx) // first call `f`, then `g`. 
// The result will be a new Context with two WriterEvents
```

can be written as
```fsharp
let f : Context -> Context = !- "f"
let g : Context -> Context = !- "g"
let combined : Context -> Context = f +> g
```

Notice that `f`, `g` and `combined` have not been executed.
Only when invoking `combined` with a `Context` argument, will the function return a new Context with two events.

```fsharp
let ctxAfter = combined Context.Default
Assert.True (ctxAfter.WriterEvents.Length = 2)
```

Maybe at some point, we should add some new unit tests in https://github.com/fsprojects/fantomas/blob/7d8b638cf25172100d60caad8be839d70d63b40c/src/Fantomas.Core.Tests/ContextTests.fs to better illustrate some of the functions used in `CodePrinter`.

(PS: @dawedawe how clear is it to you what is happening in CodePrinter?)

**Comment by dawedawe** [comment]
> (PS: @dawedawe how clear is it to you what is happening in CodePrinter?)

To use words to answer, I'd say it's mostly clear. To use numbers, I'd say it's ~85% clear.
Sometimes I'm struggling with the size of some functions/APs/expressions, which make it hard for me to have a mental model of what's going on. But that has already improved in the last couple of months. Partly due to refactoring, partly due to my growing experience and exposure to the code base.

**Comment by nojaf** [comment]
This is available in the 5.1 series.
